public/app/features/alerting/NotificationsListPage.tsx

```diff
@@ -3,29 +3,50 @@ import EmptyListCTA from 'app/core/components/EmptyListCTA/EmptyListCTA';
 import Page from 'app/core/components/Page/Page';
 import { getBackendSrv } from '@grafana/runtime';
 import { useAsyncFn } from 'react-use';
+import { appEvents } from 'app/core/core';
 import { useNavModel } from 'app/core/hooks/useNavModel';
 import { HorizontalGroup, Button, LinkButton } from '@grafana/ui';
+import { CoreEvents } from 'app/types';
 import { AlertNotification } from 'app/types/alerting';
 
-const deleteNotification = async (id: number) => {
-  return await getBackendSrv().delete(`/api/alert-notifications/${id}`);
-};
-
-const getNotifications = async () => {
-  return await getBackendSrv().get(`/api/alert-notifications`);
-};
-
 const NotificationsListPage: FC = () => {
   const navModel = useNavModel('channels');
 
   const [notifications, setNotifications] = useState<AlertNotification[]>([]);
+
+  const getNotifications = async () => {
+    return await getBackendSrv().get(`/api/alert-notifications`);
+  };
+
   const [state, fetchNotifications] = useAsyncFn(getNotifications);
   useEffect(() => {
     fetchNotifications().then(res => {
       setNotifications(res);
     });
   }, []);
 
+  const deleteNotification = (id: number) => {
+    appEvents.emit(CoreEvents.showConfirmModal, {
+      title: 'Delete',
+      text: 'Do you want to delete this notification channel?',
+      text2: `Deleting this notification channel will not delete from alerts any references to it`,
+      icon: 'trash-alt',
+      confirmText: 'Delete',
+      yesText: 'Delete',
+      onConfirm: async () => {
+        deleteNotificationConfirmed(id);
+      },
+    });
+  };
+
+  const deleteNotificationConfirmed = async (id: number) => {
+    const res = await getBackendSrv().delete(`/api/alert-notifications/${id}`);
+    fetchNotifications().then(res => {
+      setNotifications(res);
+    });
+    return res;
+  };
```

Comment:
```suggestion
  const deleteNotificationConfirmed = async (id: number) => {
    await getBackendSrv().delete(`/api/alert-notifications/${id}`);
    const notifications = await fetchNotifications();
    setNotifications(notifications);
  };
```

Reply:
nit: it might be a good idea not to mix promises and async/await syntax within the same method. It also looks like you don't need to return the `res` since you don't use it in call site.